src/sheet/sheet.c

```diff
@@ -533,14 +536,19 @@ gboolean sheet_replace (SchematicView *sv)
 	Sheet *old_sheet = schematic_view_get_sheet (sv);
 	Sheet *sheet;
 	GtkWidget *parent = gtk_widget_get_parent (GTK_WIDGET (old_sheet));
+	gdouble zoom;
 
 	g_return_val_if_fail (old_sheet != NULL, FALSE);
 	g_return_val_if_fail (IS_SHEET (old_sheet), FALSE);
 
+	sheet_get_zoom (old_sheet, &zoom);
 	sheet = SHEET (sheet_new ((double) schematic_get_width (sm) + SHEET_BORDER, (double) schematic_get_height (sm) + SHEET_BORDER));
 	if (!sheet)
 		return FALSE;
 
+	sheet_set_zoom (sheet, zoom);
+	goo_canvas_set_scale (GOO_CANVAS (sheet), zoom);
+
```

Comment:
This does not make any sense, sheet_set_zoom already calls `goo_canvas_set_scale`